package.json

```diff
@@ -41,6 +36,19 @@
     "typedoc-plugin-external-module-name": "^2.1.0",
     "typedoc-plugin-markdown": "^2.2.10"
   },
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^3.10.1",
+    "@typescript-eslint/parser": "^3.10.1",
+    "eslint": "^7.9.0",
+    "eslint-config-prettier": "^6.11.0",
+    "eslint-config-standard": "^14.1.1",
+    "eslint-plugin-jsdoc": "^29.2.0",
+    "eslint-plugin-only-warn": "^1.0.2",
+    "eslint-plugin-prettier": "^3.1.4",
+    "eslint-plugin-security": "^1.4.0",
+    "prettier": "^2.1.2",
+    "typescript": "^3.9.7"
```

Comment:
I don't think we're using this version of Typescript yet (unfortunately 😢), I suspect this is the version that Prettier is using for parsing? It would probably be best to keep this consistent with whatever is most widely used across all the libraries.

Reply:
I can remove it for now.  Maybe it is better defined per library.

Reply:
I like the idea of a `peerDependency` in each package; with the root `package.json` having a it as a `devDependency`. Right now we're pinned to 3.5.3 because 3.6 introduced some breaking changes. @joshgummersall is investigating our migration story for TypeScript.

Reply:
I think `peerDependency` is used to express "runtime dependencies" rather than build time dependencies. We don't want to express `typescript` a runtime dependency because the compiled JS we're publishing has no dependency on `typescript`. IIRC, `lerna` supports hoisted dependencies such that we _could_ list `typescript@3.5.3` in the root `package.json` and a different one in packages that differ. That's how yarn workspaces functionality works.

Reply:
npm behavior changed to not restore peerDependencies (quite awhile ago I think).  Composer is using peer dependencies to avoid including multiple different versions of a package when one has ^1.0.1 and another has ^1.0.2. We don't need to use peerDependencies for devDependencies though as peerDependenicies are only resolved through the top level dependencies list.

Reply:
We use this tool in azure-sdk-for-js to support below TS 3.6 in our public typings: https://github.com/sandersn/downlevel-dts

Reply:
@xirzec that's great new! That's exactly the tool I was looking at.